### ISSUE TITLE
Implement automatic build script instead of shell/Makefile based script

### ISSUE THREAD

**Issue by mblaschke** [webdevops/Dockerfile#140]
- [x] --no-cache option
- [x] whitelist/blacklist
- [x] auto dependency checks
- [x] automatic forced pull for dependencies based on regexp
- [x] automatic no_cache (maybe not needed after automatic forced pull)
- [x] build/push error handling
- [x] build/push retry
- [x] better python dependencies?
- [x] coding style eg https://google.github.io/styleguide/pyguide.html
- [x] fix graph dependencies (row style output of graph)

**Comment by guiyomh** [comment]
you can see the branch **feature/console**, I started writing the Makefile before holiday.
The idea was to make a command-line tools like in symfony. (but in python).

```bash
python bin/console webdevops:provision --provision=./provisioning --dockerfile=./docker --thread=3
```
or 
```bash
python bin/console webdevops:build:dockerfile --template=./template --dockerfile=./docker
```
each command are an option `-filter=?*` : tags or images name

**Comment by mblaschke** [comment]
Add automatic pull for dependencies

**Comment by mblaschke** [comment]
Released with 1.0.0
